clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -228,7 +229,8 @@ private void poll(long timeout, long now, boolean executeDelayedTasks) {
         trySend(now);
 
         // fail all requests that couldn't be sent
-        failUnsentRequests();
+        if (failUnsent)
+            failUnsentRequests();
```

Comment:
Coordinator changes are _usually_ associated with disconnects, but not necessarily. We have a `coordinatorDead()` function in `AbstractCoordinator` and I was thinking we could make a call to `client.failPendingSends(coordinator)` or something like that at the same time that we set the coordinator to null. I think that would make the behavior the same as the existing code. There may be in-flight sends to the coordinator at the time that we call `coordinatorDead()`, but that is true currently as well, and at least this would prevent any new requests from being sent.